src/main/java/marquez/service/JobService.java

```diff
@@ -242,13 +260,52 @@ public Job createOrUpdate(
     }
   }
 
+  private void updateRunInputDatasets(
+      UUID runUuid,
+      List<DatasetRow> inputRows,
+      NamespaceName namespaceName,
+      JobName jobName,
+      JobVersionRow jobVersion) {
+    if (inputRows.isEmpty()) {
+      return;
+    }
+    List<RunInput> runInputs = new ArrayList<>();
+    for (DatasetRow inputDataset : inputRows) {
+      Optional<DatasetVersionRow> version =
+          datasetVersionDao.findLatestDatasetByUuid(inputDataset.getUuid());
+      if (version.isPresent()) {
```

Comment:
I think in the existing contract those datasets have to exists. 
So we should log an error if the dataset does not exist.
In the future we may want to implicitly create those datasets.